flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java

```diff
@@ -464,22 +463,43 @@ private void finishWithSuccess(Object data) {
         }
     }
 
-    private void finishWithError(String errorCode, String errorMessage, String errorDetails) {
+    private void finishWithError(String errorCode, String errorMessage, @Nullable Exception cause) {
         if (pendingOperation != null) {
-            pendingOperation.result.error(errorCode, errorMessage, errorDetails);
+            pendingOperation.result.error(errorCode, errorMessage, createErrorMap(cause));
             pendingOperation = null;
         }
     }
 
     private void finishWithDiscoveryError(AuthorizationException ex) {
-        finishWithError(DISCOVERY_ERROR_CODE, String.format(DISCOVERY_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), getCauseFromException(ex));
+        finishWithError(DISCOVERY_ERROR_CODE, String.format(DISCOVERY_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), ex);
     }
 
     private void finishWithEndSessionError(AuthorizationException ex) {
-        finishWithError(END_SESSION_ERROR_CODE, String.format(END_SESSION_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), getCauseFromException(ex));
+        finishWithError(END_SESSION_ERROR_CODE, String.format(END_SESSION_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), ex);
+    }
+
+    private Map<String, Object> createErrorMap(@Nullable Exception cause) {
```

Comment:
Just spotted this and may explain why there was a misunderstanding on what I wrote below. Prior to your PR, I was trying to extract extra information from the cause of the exception. The exception itself (note: not the cause) would most likely be an instance of an `AuthorizationException`. Your PR now changes it to look at the exception that's been thrown, not the cause. However, what's passed here isn't actually the cause of the exception but the exception itself. The name of the parameter is named `cause`. Perhaps this one of the factors that caused some confusion on which level of the exception/error hierarchy you're looking at

Reply:
Realised there was a lot of text there but the summary/tl;dr is IMO better to rename `cause` to avoid confusion

Reply:
Done :)